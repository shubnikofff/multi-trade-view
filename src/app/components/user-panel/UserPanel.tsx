import React from 'react';

import { useAuth } from '@hooks';

import './UserPanel.scss';

function UserPanel() {
    const { auth, switchUser } = useAuth()

    return (
        <div className="user-panel">
            <div>
                <span>You are <b>{auth}</b></span>
            </div>
            <div>
                <button className="btn btn-outline" onClick={switchUser}>
                    Switch user
                </button>
            </div>
        </div>
    );
}

export default UserPanel;
