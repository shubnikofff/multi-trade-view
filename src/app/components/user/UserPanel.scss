.user-panel {
  display: flex;
  justify-content: space-between;
  align-items: baseline;
}
