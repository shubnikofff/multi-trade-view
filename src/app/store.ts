import {
    configureStore,
    createSerializableStateInvariantMiddleware,
    isPlain,
} from '@reduxjs/toolkit';

import { reducer as authReducer } from './slices/authSlice';
import { reducer as chatsReducer } from './slices/chatsSlice';
import { reducer as tradesReducer } from './slices/tradesSlice';
import { reducer as usersReducer } from './slices/usersSlice';

const serializableMiddleware = createSerializableStateInvariantMiddleware({
    isSerializable: value => isPlain(value) || value instanceof Date,
})

const store = configureStore({
    reducer: {
        auth: authReducer,
        chats: chatsReducer,
        trades: tradesReducer,
        users: usersReducer,
    },
    middleware: [serializableMiddleware],
});

export type RootState = ReturnType<typeof store.getState>

export default store;
