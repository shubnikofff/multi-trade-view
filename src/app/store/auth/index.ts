import slice from './authSlice';

export const {
    actions,
    reducer,
} = slice;
