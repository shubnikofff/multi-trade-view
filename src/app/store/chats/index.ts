import slice from './chatsSlice';
import selectors from './chatsSelectors';

export const {
    actions,
    reducer,
} = slice;

export {
    selectors,
};
