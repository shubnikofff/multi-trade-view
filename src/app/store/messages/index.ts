import slice from './messagesSlice';
import selectors from './messagesSelectors';

export const {
    actions,
    reducer,
} = slice;

export {
    selectors,
}
