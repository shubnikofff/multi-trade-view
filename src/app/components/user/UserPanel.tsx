import React from 'react';

import { Grid, Button } from '@material-ui/core';

import { useUserPanel } from './useUserPanel';
import { UserRole } from '../../types/user';

interface UserPanelProps {
}

function UserPanel() {
    const { auth, switchUser } = useUserPanel()

    return (
        <Grid
            container
            justify="space-between"
            alignItems="center"
        >
            <Grid item>
                You are <b>{auth === UserRole.Seller ? 'Seller' : 'Buyer'}</b>
            </Grid>
            <Grid item>
                <Button
                    color="primary"
                    variant="outlined"
                    onClick={switchUser}
                >
                    Switch user
                </Button>
            </Grid>
        </Grid>
    );
}

export default UserPanel;
