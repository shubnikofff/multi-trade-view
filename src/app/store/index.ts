import {
    configureStore,
    createSerializableStateInvariantMiddleware,
    isPlain,
} from '@reduxjs/toolkit';

import { reducer as authReducer } from './auth';
import { reducer as chatsReducer } from './chats';
import { reducer as messagesReducer } from './messages';
import { reducer as tradesReducer } from './trades';
import { reducer as usersReducer } from './users';

const serializableMiddleware = createSerializableStateInvariantMiddleware({
    isSerializable: value => isPlain(value) || value instanceof Date,
});

const store = configureStore({
    reducer: {
        auth: authReducer,
        chats: chatsReducer,
        messages: messagesReducer,
        trades: tradesReducer,
        users: usersReducer,
    },
    middleware: [serializableMiddleware],
});

export default store;
