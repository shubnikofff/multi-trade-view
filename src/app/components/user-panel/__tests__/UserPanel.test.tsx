import React from 'react';

import store from '../../../store';
import { mount } from 'enzyme';

import { Provider } from 'react-redux';
import UserPanel from '../UserPanel';

describe('Component UserPanel', () => {
    const wrapper = mount(
        <Provider store={store}>
            <UserPanel />
        </Provider>
    );

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should switch user to buyer', () => {
        wrapper.find('button').simulate('click');

        expect(wrapper).toMatchSnapshot();
    });
});
