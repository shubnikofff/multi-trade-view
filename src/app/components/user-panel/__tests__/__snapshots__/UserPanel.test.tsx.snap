// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component UserPanel should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <UserPanel>
    <div
      className="user-panel"
    >
      <div>
        <span>
          You are 
          <b>
            seller
          </b>
        </span>
      </div>
      <div>
        <button
          className="btn btn-outline"
          onClick={[Function]}
        >
          Switch user
        </button>
      </div>
    </div>
  </UserPanel>
</Provider>
`;

exports[`Component UserPanel should switch user to buyer 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <UserPanel>
    <div
      className="user-panel"
    >
      <div>
        <span>
          You are 
          <b>
            buyer
          </b>
        </span>
      </div>
      <div>
        <button
          className="btn btn-outline"
          onClick={[Function]}
        >
          Switch user
        </button>
      </div>
    </div>
  </UserPanel>
</Provider>
`;
